catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java

```diff
@@ -0,0 +1,569 @@
+        console.println("Number of metacards exported: " + exportedItems.size());
+        console.println();
+
+        console.println("Starting content export...");
+        start = Instant.now();
+        List<ExportItem> exportedContentItems = doContentExport(zipFile, exportedItems);
+        console.println("Content exported in: " + Duration.between(start, Instant.now()));
+        console.println("Number of content exported: " + exportedContentItems.size());
+
+        console.println();
+
+        if (delete) {
+            doDelete(exportedItems, exportedContentItems);
+        }
+
+        console.println("Signing zip file...");
+        start = Instant.now();
+        jarSigner.signJar(zipFile.getFile(),
+                System.getProperty("org.codice.ddf.system.hostname"),
+                System.getProperty("javax.net.ssl.keyStorePassword"),
+                System.getProperty("javax.net.ssl.keyStore"),
+                System.getProperty("javax.net.ssl.keyStorePassword"));
+        console.println("zip file signed in: " + Duration.between(start, Instant.now()));
+        console.println("Export complete.");
+        return null;
+    }
+
+    private File initOutputFile(String output) {
+        String resolvedOutput;
+        File initialOutputFile = new File(output);
+        if (initialOutputFile.isDirectory()) {
+            // If directory was specified, auto generate file name
+            resolvedOutput = Paths.get(initialOutputFile.getPath(), name.get())
+                    .toString();
+        } else {
+            resolvedOutput = output;
+        }
+
+        return new File(resolvedOutput);
+    }
+
+    private List<ExportItem> doMetacardExport(ZipFile zipFile, Filter filter) {
+        List<ExportItem> exportedItems = new ArrayList<>();
+        for (Result result : new QueryResulterable(catalogFramework,
+                (i) -> getQuery(filter, i, PAGE_SIZE),
+                PAGE_SIZE)) {
+            writeToZip(zipFile, result);
+            exportedItems.add(new ExportItem(result.getMetacard()
+                    .getId(),
+                    getTag(result),
+                    result.getMetacard()
+                            .getResourceURI()));
+            // Fetch and export all history for each exported item
+            for (Result revision : new QueryResulterable(catalogFramework,
+                    (i) -> getQuery(getHistoryFilter(result), i, PAGE_SIZE),
+                    PAGE_SIZE)) {
+                writeToZip(zipFile, revision);
+                exportedItems.add(new ExportItem(revision.getMetacard()
+                        .getId(),
+                        getTag(revision),
+                        revision.getMetacard()
+                                .getResourceURI()));
+            }
+        }
+        return exportedItems;
+    }
+
+    private List<ExportItem> doContentExport(/*Mutable,IO*/ZipFile zipFile,
```

Comment:
Yup it does both. The list of items from the metacard export contains the history and original metacards